backends/dnf/meson.build

```diff
@@ -25,7 +25,7 @@ install_data(
 )
 
 add_languages('cpp', native: false)
-shared_module(
+shared_library(
```

Comment:
Does this still produce an unversioned library?

Reply:
Looks like it did, yes.

My reasoning for switching to shared_library is that we can now enforce that there are no missing symbols.
But `shared_module` is still more semantically correct I suppose: this is, afterall, a loadable module.
I've reverted back.